packages/client/hooks/useTrebuchetEvents.ts

```diff
@@ -63,6 +63,7 @@ const useTrebuchetEvents = () => {
       // hacky but that way we don't have to double parse huge json payloads. SSE graphql payloads are pre-parsed
       if (typeof payload !== 'string' || !payload.startsWith('{"version":')) return
```

Comment:
Payload is always an object rather than a string for me but it does contain the version. Perhaps this is the wrong place to update atmosphere (if at all) or perhaps there's a difference in my set-up in dev vs prod so that I'm receiving an object rather than a string

Reply:
looks like this bit of code is outdated.

the service worker gets updated in 1 of 2 ways:
- in a window onload event when `register` gets called
- in this bit of code, which as you've found out is currently broken!

to trigger the `controllerchange` event, the sw.js file fetched from the server has to be different than what is cached. 

i'd suggest we change this to something like
```ts
      // should always be of type object
      const obj = typeof payload === 'string' ? JSON.parse(payload) : payload
      if (obj.version !== __APP_VERSION__ && 'serviceWorker' in navigator) {
```